src/crimson/osd/osd_operations/pg_advance_map.h

```diff
@@ -34,11 +34,12 @@ class PGAdvanceMap : public PhasedOperationT<PGAdvanceMap> {
 
   PeeringCtx rctx;
   const bool do_init;
+  const bool split_child;
```

Comment:
Can you explain why should PGAdvanceMap know it's a child pg? Perhaps the earlier comments regarding the additional call to PGAdvanceMap from PGSplitting will resolve the issue that is being solved here (and we will be able to avoid the additional member)?